infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationConfigManager.java

```diff
@@ -86,28 +85,31 @@ public LogReplicationConfigManager(CorfuRuntime runtime, String pluginConfigFile
 
     private void openVersionTable() {
         try {
-            pluginVersionTable = corfuStore.openTable(CORFU_SYSTEM_NAMESPACE,
-                    LOG_REPLICATION_PLUGIN_VERSION_TABLE, VersionString.class,
-                    Version.class, Uuid.class, TableOptions.builder().build());
+            pluginVersionTable = corfuStore.openTable(CORFU_SYSTEM_NAMESPACE, LOG_REPLICATION_PLUGIN_VERSION_TABLE,
+                VersionString.class, Version.class, Uuid.class, TableOptions.builder().build());
         } catch (NoSuchMethodException | InvocationTargetException | IllegalAccessException e) {
             log.warn("Exception when opening config tables", e);
             throw new UnrecoverableCorfuError(e);
         }
     }
 
-    public Set<String> getStreamsToReplicate() {
+    public Map<ReplicationSubscriber, Set<String>> getSubscriberToStreamsMap() {
         try {
             currentVersion = logReplicationConfigAdapter.getVersion();
-            Set<String> streams = logReplicationConfigAdapter.fetchStreamsToReplicate();
-            // Add registryTable to the streams
-            String registryTable = getFullyQualifiedTableName(
-                    CORFU_SYSTEM_NAMESPACE, TableRegistry.REGISTRY_TABLE_NAME);
-            streams.add(registryTable);
+            Map<ReplicationSubscriber, Set<String>> subscriberToStreamsMap =
+                logReplicationConfigAdapter.getSubscriberToStreamsMap();
```

Comment:
Just to confirm: this is just a placeholder until the dynamic streams PR is in place, right?

Reply:
actually, with the dynamic streams PR, getStreamsToReplicate() will not be an interface in the plugin.  I was thinking that we can have this new method(completely internal, not in the plugin) which gets everything in 1 scan.